homeassistant/components/fing/config_flow.py

```diff
@@ -0,0 +1,100 @@
+"""Config flow file."""
+
+from typing import Any
+
+from fing_agent_api import FingAgent
+import httpx
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant
+
+from .const import AGENT_IP, AGENT_KEY, AGENT_PORT, DOMAIN
+
+
+def _get_data_schema(
+    hass: HomeAssistant, config_entry: ConfigEntry | None = None
+) -> vol.Schema:
+    """Get a schema with default values."""
+
+    if config_entry is None:
+        return vol.Schema(
+            {
+                vol.Required(CONF_NAME, default="Fing Agent"): str,
+                vol.Required(AGENT_IP): str,
+                vol.Required(AGENT_PORT, default="49090"): str,
+                vol.Required(AGENT_KEY): str,
+            }
+        )
+
+    return vol.Schema(
+        {
+            vol.Required(CONF_NAME, default=config_entry.data.get(CONF_NAME)): str,
+            vol.Required(AGENT_IP, default=config_entry.data.get(AGENT_IP)): str,
+            vol.Required(AGENT_PORT, default=config_entry.data.get(AGENT_PORT)): str,
+            vol.Required(AGENT_KEY, default=config_entry.data.get(AGENT_KEY)): str,
+        }
+    )
+
```

Comment:
entry is never none, so this can be inlined

Reply:
I've adjusted the general flow of this section. 
The **config_entry** parameter was used in an older implementation (before there was an implementation of the options flow).